src/components/tools/NewToolRequestDialog.js

```diff
@@ -0,0 +1,180 @@
+                                name="source_url"
+                                label={t("toolSrcLinkLabel")}
+                                required={true}
+                                margin="dense"
+                                validate={validateUrl}
+                                id={build(baseId, ids.TOOL_REQUEST.SRC_LINK)}
+                                component={FormTextField}
+                            />
+                            <Field
+                                name="version"
+                                label={t("toolVersionLabel")}
+                                required={true}
+                                margin="dense"
+                                id={build(baseId, ids.TOOL_REQUEST.VERSION)}
+                                component={FormTextField}
+                            />
+                            <Field
+                                name="documentation_url"
+                                label={t("toolDocumentationLabel")}
+                                required={true}
+                                margin="dense"
+                                id={build(
+                                    baseId,
+                                    ids.TOOL_REQUEST.DOCUMENTATION
+                                )}
+                                validate={validateUrl}
+                                component={FormTextField}
+                            />
+                            <Field
+                                name="cmd_line"
+                                label={t("toolInstructionsLabel")}
+                                required={false}
+                                margin="dense"
+                                id={build(
+                                    baseId,
+                                    ids.TOOL_REQUEST.INSTRUCTIONS
+                                )}
+                                component={FormMultilineTextField}
+                            />
+                            <Field
+                                name="test_data_path"
+                                label={t("toolTestDataLabel")}
+                                required={false}
+                                margin="dense"
+                                id={build(
+                                    baseId,
+                                    ids.TOOL_REQUEST.TEST_DATA_LINK
+                                )}
+                                component={FormTextField}
+                            />
+                            <Button
+                                style={{ float: "right" }}
+                                color="primary"
+                                type="submit"
+                                id={build(baseId, ids.BUTTONS.SUBMIT)}
+                                aria-label={t("submit")}
+                            >
+                                {t("submit")}
+                            </Button>
```

Comment:
Is this submit button within the form? I would think this would be within `DialogActions` so it's always visible and doesn't scroll

Reply:
Hmm I have tried it earlier and I tried it again now but putting the `Submit` button outside doesn't seem to be working. I am newbie when it comes `Formik` so I may be missing something.

Reply:
Ah, I'm mixing up words - when I said `form` I meant the dialog content, not the `Form` tag.  At least for [CreateFolderDialog](https://github.com/cyverse-de/sonora/blob/eae296849154c04c1f2b6a4a58fe7db65b5e7b2c/src/components/data/CreateFolderDialog.js#L74-L86), I put the entire dialog within `Form`.  Though, Paul may know a different way to do this.

Reply:
Ahh ok. Even then since I am using `DEDialog`, I will have to pass it as `actions` prop and that doesn't seem to work 😩

Reply:
As far as I know, the submit button has to be inside the `Form`, but wrapping the entire dialog in the `Form` should work 👍

Reply:
Ah, yeah, using `DEDialog` makes this trickier 🤔

Reply:
Yup. May `DEDialog` shouldn't be used with `Formik` or should we refactor `DEDialog` to include `DialogActions` as children 🤔

Reply:
Does this component work?  It's using `DEDialog` with Formik: https://github.com/cyverse-de/sonora/blob/master/src/components/data/SaveAsDialog.js

Reply:
Right now the `NewToolRequestDialog` is return something like
```jsx
return (
    <DEDialog ...>
        <Formik ...>
            <Form>...</Form>
        </Formik>
    </DEDialog>
);
```
But this should work too:
```jsx
return (
    <Formik ...>
        <Form ...>
            <DEDialog>...</DEDialog>
        </Form>
    </Formik>
);
```

Reply:
@ash3rz Jinx! That's pretty much what I was just thinking 👍

Reply:
Ok got it to work. You have to call `handleSubmit` `onClick` explicitly with this approach. Thanks @ash3rz @psarando.

Reply:
I will probably do the same for https://github.com/cyverse-de/sonora/blob/master/src/components/tools/edit/EditTool.js form as well .